File: Design and Programming Environments/Iteration 2/TicketManagerCSharp/TicketManagerCSharp/repository/IRepository.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using TicketManagerCSharp.domain;

namespace TicketManagerCSharp.repository
{
    public interface IRepository<ID, E> where E : HasId<ID>
    {
        E FindOne(ID id);

        IEnumerable<E> FindAll();

        void Save(E entity);

        void Delete(ID id);

        void Update(ID id, E entity);

        int Size();
    }
}

File: Second year/Design and Programming Environments/TicketManagerJava/Persistance/src/main/java/Repository/IRepository.java
package Repository;

import ModelDomain.IHasID;

public interface IRepository<ID,T extends IHasID<ID>> {
    int size();
    void save(T entity);
    void delete(ID id);
    void update(ID id, T entity);
    T findOne(ID id);
    Iterable<T> findAll();
}
